hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
 	}
 
 	private void checkTransactionNeeded() {
-		if ( !isTransactionInProgress() ) {
+		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
```

Comment:
i have just moved the code into the constructor and call the instance variable  disallowOutOfTransactionUpdateOperations, it seems a little more readable not negation.

@sebersole should I revert it or it's fine?